alibi_detect/tests/test_datasets.py

```diff
@@ -1,20 +1,128 @@
+def test_types_cifar10c():
+    assert len(corruption_list) == 19
+
+
+@pytest.mark.parametrize('return_X_y', [True, False])
+def test_fetch_cifar10c(return_X_y):
+    corruption = list(np.random.choice(corruption_list, 5, replace=False))
+    try:
+        data = fetch_cifar10c(corruption=corruption, severity=2, return_X_y=return_X_y)
+    except RequestException:
+        pytest.skip('CIFAR-10-C dataset URL down')
+    if return_X_y:
+        assert isinstance(data, tuple)
+        assert isinstance(data[0], np.ndarray) and isinstance(data[1], np.ndarray)
+    else:
+        assert isinstance(data, Bunch)
+        assert isinstance(data.data, np.ndarray) and isinstance(data.target, np.ndarray)
+
+
+# Attack datasets
+datasets = ['cifar10']
+models = ['resnet56']
+attacks = ['cw', 'slide']
+
+
+@pytest.mark.parametrize('return_X_y', [True, False])
+def test_fetch_cifar10c(return_X_y):
+    dataset = list(np.random.choice(datasets, 1))[0]
+    model = list(np.random.choice(models, 1))[0]
+    attack = list(np.random.choice(attacks, 1))[0]
+    try:
+        data = fetch_attack(dataset=dataset, model=model, attack=attack, return_X_y=return_X_y)
+    except RequestException:
+        pytest.skip('Attack dataset URL down for dataset %s, model %s, and attack %s' % (dataset, model, attack))
+    if return_X_y:
+        assert isinstance(data, tuple)
+        assert isinstance(data[0][0], np.ndarray) and isinstance(data[0][1], np.ndarray) and \
+               isinstance(data[1][0], np.ndarray) and isinstance(data[1][1], np.ndarray)
+    else:
+        assert isinstance(data, Bunch)
+        assert isinstance(data.data_train, np.ndarray) and isinstance(data.data_test, np.ndarray) and \
+               isinstance(data.target_train, np.ndarray) and isinstance(data.target_test, np.ndarray)
+        assert data.meta['attack_type'] == attack
+
+
+# NAB dataset
 files = get_list_nab()
 
 
 def test_list_nab():
     assert len(files) == 58
 
 
-def test_fetch_nab():
+@pytest.mark.parametrize('return_X_y', [True, False])
+def test_fetch_nab(return_X_y):
     idx = np.random.choice(len(files))
-    data = fetch_nab(files[idx])
-    assert isinstance(data, Bunch)
-    assert isinstance(data.data, pd.DataFrame) and isinstance(data.target, pd.DataFrame)
-    data = fetch_nab(files[idx], return_X_y=True)
-    assert isinstance(data, tuple)
-    assert isinstance(data[0], pd.DataFrame) and isinstance(data[1], pd.DataFrame)
+    try:
+        data = fetch_nab(files[idx], return_X_y=return_X_y)
+    except RequestException:
+        pytest.skip('NAB dataset URL down')
+    if return_X_y:
+        assert isinstance(data, tuple)
+        assert isinstance(data[0], pd.DataFrame) and isinstance(data[1], pd.DataFrame)
+    else:
+        assert isinstance(data, Bunch)
+        assert isinstance(data.data, pd.DataFrame) and isinstance(data.target, pd.DataFrame)
+
+
+# Genome dataset
+# TODO - Genome dataset is large compared to others - do we want to include in regular CI?
```

Comment:
Probably not, some notebooks are excluded for the same reason.

If we did we would need to use our own CI image with all the data pre-loaded, but then testing the fetching part is a moot point as it's already available locally.

Reply:
Fair point. The current tests only add on ~90 secs (locally) so maybe an OK compromise? We could maybe add slowly ones to non-nightly CI when we address #322?

Reply:
I'll remove TODO.

Reply:
I'd keep the TODO as it keeps a record of the discussion in the source.